From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Tue, 10 Feb 2026 13:31:44 -0500
Subject: [PATCH] Chlorophyll Disable Default Puferfish Configs


diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
index 4e80cbfd54d94321b164556e7cf011ef9193e2e5..d0acbb37c7b19b7d710fcd26db55d1f280eaaf2c 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
@@ -73,7 +73,7 @@ public class PufferfishConfig extends StaticConfig {
         "be left enabled on most servers, but is provided as a",
         "configuration option if the vanilla deviation is undesirable."
     })
-    public static boolean ENABLE_SUFFOCATION_OPTIMIZATION = true;
+    public static boolean ENABLE_SUFFOCATION_OPTIMIZATION = false;
 
     @Priority(4)
     @Comment({
@@ -83,9 +83,9 @@ public class PufferfishConfig extends StaticConfig {
         "One quick note - this does not actually spawn mobs async (that would be very unsafe).",
         "This just offloads some expensive calculations that are required for mob spawning."
     })
-    public static boolean ENABLE_ASYNC_MOB_SPAWNING = true;
+    public static boolean ENABLE_ASYNC_MOB_SPAWNING = false;
     @Ignore
-    public static boolean _ENABLE_ASYNC_MOB_SPAWNING = true;
+    public static boolean _ENABLE_ASYNC_MOB_SPAWNING = false;
     @Ignore
     public static boolean ASYNC_MOB_SPAWNING_INITIALIZED = false;
 
@@ -114,7 +114,7 @@ public class PufferfishConfig extends StaticConfig {
     })
     public static class DAB {
 
-        public static boolean ENABLED = true;
+        public static boolean ENABLED = false;
 
         @Comment({
             "This value determines how far away an entity has to be",
@@ -177,7 +177,7 @@ public class PufferfishConfig extends StaticConfig {
         "Throttles the AI goal selector in entity inactive ticks.",
         "This can improve performance by a few percent, but has minor gameplay implications."
     })
-    public static boolean INACTIVE_GOAL_SELECTOR_THROTTLE = true;
+    public static boolean INACTIVE_GOAL_SELECTOR_THROTTLE = false;
 
     @Priority(11)
     @Comment({
