From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Tue, 10 Feb 2026 13:37:44 -0500
Subject: [PATCH] Chlorophyll Toggle Pufferfish Optimizations


diff --git a/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java b/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
index ff831db0cb16f130ad702677f6805051debeb0a8..cfc84a5fb95499a6022eee95f255e9a1a8fc1257 100644
--- a/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
+++ b/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
@@ -119,6 +119,7 @@ public abstract class MoveToBlockGoal extends Goal {
                 for (int x = 0; x <= r; x = x > 0 ? -x : 1 - x) {
                     for (int z = x < r && x > -r ? r : 0; z <= r; z = z > 0 ? -z : 1 - z) {
                         pos.setWithOffset(mobPos, x, y - 1, z);
+                        if (me.biquaternions.chlorophyll.ChlorophyllConfig.TOGGLES.PUFFERFISH.BLOCK_GOAL_CHUNK_LOADING_OPTIMIZATION) // Chlorophyll - Make toggleable
                         if (!this.mob.level().hasChunkAt(pos)) continue; // Pufferfish - if this block isn't loaded, continue
                         if (this.mob.isWithinHome(pos) && this.isValidTarget(this.mob.level(), pos)) {
                             this.blockPos = pos;
diff --git a/net/minecraft/world/entity/monster/EnderMan.java b/net/minecraft/world/entity/monster/EnderMan.java
index 6cd88eff2ac2583488cd0701aef497e9a3fc85b8..19652fce6476ffde73a44baf84e12bb84387a63f 100644
--- a/net/minecraft/world/entity/monster/EnderMan.java
+++ b/net/minecraft/world/entity/monster/EnderMan.java
@@ -299,7 +299,7 @@ public class EnderMan extends Monster implements NeutralMob {
         BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos(x, y, z);
 
         // Pufferfish start - single chunk lookup
-        net.minecraft.world.level.chunk.LevelChunk chunk = this.level().getChunkIfLoaded(pos);
+        net.minecraft.world.level.chunk.LevelChunk chunk = this.level().getChunkIfLoaded(pos); // Chlorophyll - Keep in check, it might damage RNG manipulation (enderman teleport)
         if (chunk == null) {
             return false;
         }
diff --git a/net/minecraft/world/entity/projectile/Projectile.java b/net/minecraft/world/entity/projectile/Projectile.java
index 199ae275d13903e7499d07eaf0bb6b898863b8f3..e40542310d1cda11aed0f63dddf42322a7c97320 100644
--- a/net/minecraft/world/entity/projectile/Projectile.java
+++ b/net/minecraft/world/entity/projectile/Projectile.java
@@ -57,6 +57,11 @@ public abstract class Projectile extends Entity implements TraceableEntity {
     private int pufferfish$loadedLifetime = 0;
     @Override
     public void setPos(final double x, final double y, final double z) {
+        // Chlorophyll start - Toggle Reduce Projectile Chunk Loading
+        if (!me.biquaternions.chlorophyll.ChlorophyllConfig.TOGGLES.PUFFERFISH.PROJECTILE_CHUNK_LOADING_OPTIMIZATION) {
+            super.setPos(x, y, z);
+            return;
+        } // Chlorophyll stop - Toggle Reduce Projectile Chunk Loading
         int currentTick = net.minecraft.server.MinecraftServer.currentTick;
         if (Projectile.PUFFERFISH$LOADED_TICK != currentTick) {
             Projectile.PUFFERFISH$LOADED_TICK = currentTick;
